include/fs/fs.h

```diff
@@ -294,15 +300,15 @@ ssize_t fs_read(struct fs_file_t *zfp, void *ptr, size_t size);
  * @param ptr Pointer to the data buffer
  * @param size Number of bytes to be write
  *
- * @return Number of bytes written. On success, it will be equal to the number
- * of bytes requested to be written. Any other value, indicates an error. Will
- * return -ERRNO code on error.
- * In the case where -ERRNO is returned, the file pointer will not be
+ * @retval >=0 number of bytes written, on success;
+ * @retval <0 negative errno code on error.
+ *
+ * @return In the case where error is returned, the file pointer will not be
```

Comment:
I generally prefer to see descriptions like this and the one for `fs_read` up in the description of how the function behaves, and in the return value summary just have `count of bytes written` maybe with `as described above` if that doesn't seem enough.